src/main/java/uk/gov/ons/ctp/response/casesvc/client/ActionSvcClient.java

```diff
@@ -0,0 +1,96 @@
+package uk.gov.ons.ctp.response.casesvc.client;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.util.List;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponents;
+import uk.gov.ons.ctp.common.rest.RestUtility;
+import uk.gov.ons.ctp.response.action.representation.ActionDTO;
+import uk.gov.ons.ctp.response.action.representation.ActionPlanDTO;
+import uk.gov.ons.ctp.response.casesvc.config.AppConfig;
+
+/** The impl of the service which calls the action service via REST */
+@Service
+public class ActionSvcClient {
+  private static final Logger log = LoggerFactory.getLogger(ActionSvcClient.class);
+
+  @Autowired private AppConfig appConfig;
+
+  @Autowired private RestTemplate restTemplate;
+
+  @Qualifier("actionServiceRestUtility")
+  @Autowired
+  private RestUtility restUtility;
+
+  /**
+   * Make use of the ActionService to create and post a new Action for a given caseId according to
+   * Category actionType and CaseEvent createdBy values
+   *
+   * @param actionType action type
+   * @param caseId the UUID caseId
+   * @param createdBy who did this
+   */
+  public void createAndPostAction(String actionType, UUID caseId, String createdBy) {
```

Comment:
breaking explicitely in naming convention the SRP !!!!! A method don not do 2 things . If you need a ActionDTO mayebe you should pass to the contract ? I know it s a small thing but looks bad.